pkg/kubelet/types/constants.go

```diff
@@ -16,8 +16,22 @@ limitations under the License.
 
 package types
 
+import "runtime"
+
 const (
-	PodInfraContainerImage = "gcr.io/google_containers/pause:2.0"
+
 	// system default DNS resolver configuration
 	ResolvConfDefault = "/etc/resolv.conf"
+
+	DefaultPodInfraContainerImageName = "gcr.io/google_containers/pause"
+	DefaultPodInfraContainerImageVersion = "2.0"
 )
+
+// Returns the arch-specific pause image that kubelet use as the default
+func GetDefaultPodInfraContainerImage() string {
+	if runtime.GOARCH == "amd64" {
+		return DefaultPodInfraContainerImageName + ":" + DefaultPodInfraContainerImageVersion
+	} else {
+		return DefaultPodInfraContainerImageName +  "-" + runtime.GOARCH + ":" + DefaultPodInfraContainerImageVersion
```

Comment:
Yeah, it's work in progress. Images like `etcd`, `hyperkube`, `kube-proxy` are turned into `etcd-amd64`, `hyperkube-amd64` and `kube-proxy-amd64`. The older naming convention (`hyperkube` is kept for backwards-compability for a while though)